src/osd/strconv.cpp

```diff
@@ -12,25 +12,125 @@
+	};
+
+	// implementation of string_source for NUL-terminated strings
+	template<typename T>
+	class null_terminated_string_source : public string_source<T>
+	{
+	public:
+		null_terminated_string_source(const T *str) : m_str(str)
+		{
+			assert(str != nullptr);
+		}
+
+		virtual const T *string() const override { return m_str; }
+		virtual int char_count() const override { return -1; }
+
+	private:
+		const T *m_str;
+	};
+
+	// implementation of string_source for std::[w]string
+	template<typename T>
+	class basic_string_source : public string_source<T>
+	{
+	public:
+		basic_string_source(const std::basic_string<T> &str) : m_str(str)
+		{
+		}
+
+		virtual const T *string() const override { return m_str.c_str(); }
+		virtual int char_count() const override { return (int) m_str.size() + 1; }
+
+	private:
+		const std::basic_string<T> &m_str;
+	};
+};
+
+//============================================================
+//  mbstring_from_wstring
+//============================================================
+
+static std::string &mbstring_from_wstring(std::string &dst, UINT code_page, const string_source<wchar_t> &src)
+{	
+	// convert UTF-16 to the specified code page
+	int dst_char_count = WideCharToMultiByte(code_page, 0, src.string(), src.char_count(), nullptr, 0, nullptr, nullptr);
+	dst.resize(dst_char_count - 1);
+	WideCharToMultiByte(code_page, 0, src.string(), src.char_count(), &dst[0], dst_char_count, nullptr, nullptr);
+
+	return dst;
+}
+
+
+//============================================================
+//  wstring_from_mbstring
+//============================================================
+
+static std::wstring &wstring_from_mbstring(std::wstring &dst, const string_source<char> &src, UINT code_page)
+{
+	// convert multibyte string (in specified code page) to UTF-16
+	int dst_char_count = MultiByteToWideChar(code_page, 0, src.string(), src.char_count(), nullptr, 0);
+	dst.resize(dst_char_count - 1);
+	MultiByteToWideChar(CP_UTF8, 0, src.string(), src.char_count(), &dst[0], dst_char_count - 1);
+
+	return dst;
+}
+
+
+//============================================================
+//  astring_from_utf8
+//============================================================
+
+std::string &astring_from_utf8(std::string &dst, const std::string &s)
+{
+	// convert MAME string (UTF-8) to UTF-16
+	std::wstring wstring = wstring_from_utf8(s);
+
+	// convert UTF-16 to "ANSI code page" string
+	return mbstring_from_wstring(dst, CP_ACP, basic_string_source<wchar_t>(wstring));
+}
+
+
+
 //============================================================
 //  astring_from_utf8
 //============================================================
 
 std::string &astring_from_utf8(std::string &dst, const char *s)
 {
 	// convert MAME string (UTF-8) to UTF-16
-	std::basic_string<WCHAR> wstring = wstring_from_utf8(s);
+	std::wstring wstring = wstring_from_utf8(s);
```

Comment:
I have not been able to find an "official" declaration, but I'm 99.99% sure that it is.  If you look at the Microsoft CRT source code, wchar_t and WCHAR are used interchangeably, whether within code or implicitly in calls to Win32.